woocommerce/woocommerce-android
# Integrate UI with the Stats V4 Top Performers API

Summary
==========
Fixes issue woocommerce/woocommerce-admin#2538, this PR makes the entire `MyStoreFragment` compliant with the Stats V4 Top Performers API. Replacing the current usage of the `WCTopEarnerModel` of the `MyStoreDashboard` with the new `WCTopPerformerProductModel` as implemented on issue woocommerce/woocommerce-admin#2537. Also, this PR improves the usage of the offline capabilities of the app when retrieving the Top Performers data and introduces the usage of coroutines on the `MyStorePresenter`.

Changelog
==========
| Description | Commit |
| ------------- | ------------- |
| Migrates the current implementation of the Top Earners to the Top Performers from Stats V4 API | 7d832986b718ab213ce3e31abfaa363b44fbbe53 | 
| Fix the bug introduced on the last commit caused by changing the wrong fetch call | c6aca79bbc0ff113417adc415e38211823e35b17 | 
| Make the Top Performers call capable of using the forced call handling | 88cfe3ac63a20df22e1548e0f02440f2de363f70 | 
| Improve suspended functions definitions around the `MyStorePresenter` | 834368653305e0923c7359ddfa06ff4542d3bb5c | 
| Adapt the `MyStorePresenter` tests to work with the new infrastructure around coroutines | 380f32254b7878c5173fe912fb035ff8331d55d9 | 

Screenshots
==========
| When Top Performers data available | When there's no Top Performers data available |
| ------------- | ------------- |
| ![Screenshot_1596776373](https://user-images.githubusercontent.com/5920403/89611441-9e6c3c00-d853-11ea-988c-7273d87e4cc9.png) | ![Screenshot_1596776378](https://user-images.githubusercontent.com/5920403/89611443-a035ff80-d853-11ea-98da-8b88036fc958.png) |

How to Test
==========
1. Log in with a store running V4 Woo and *does have* WCAdmin installed.
2. On the My Store page make sure that the Stats V4 Top Performers is showing up and loading the stats synced with the Woo Site data.
3. Click on a Top Performer Product and verify if the Product Details View is open and works as it should.

Update release notes:
